src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -99,7 +99,7 @@ internal OpenApiInfo GetOpenApiInfo()
     /// the object to support filtering each
     /// description instance into its appropriate document.
     /// </remarks>
-    private OpenApiPaths GetOpenApiPaths(HashSet<OpenApiTag> capturedTags)
+    private async Task<OpenApiPaths> GetOpenApiPathsAsync(HashSet<OpenApiTag> capturedTags, CancellationToken cancellationToken = default)
```

Comment:
Is making all these private methods have the cancellation token be optional just setting the code up for a future edit to accidentally forget to pass it along? Should they just all be required except for on any public methods?